x/batching/keeper/abci.go

```diff
@@ -52,69 +52,68 @@ func (k Keeper) ConstructBatch(ctx sdk.Context) (types.Batch, error) {
 	if err != nil {
 		return types.Batch{}, err
 	}
-	dataRootHex, err := k.ConstructDataResultTree(ctx)
+	dataLeaves, dataRootHex, err := k.ConstructDataResultTree(ctx)
 	if err != nil {
 		return types.Batch{}, err
 	}
-	valRootHex, err := k.ConstructValidatorTree(ctx)
+	valLeaves, valRootHex, err := k.ConstructValidatorTree(ctx)
 	if err != nil {
 		return types.Batch{}, err
 	}
 
 	return types.Batch{
-		BatchNumber:    curBatchNum,
-		BlockHeight:    ctx.BlockHeight(),
-		DataResultRoot: dataRootHex,
-		ValidatorRoot:  valRootHex,
-		BlockTime:      ctx.BlockTime(),
+		BatchNumber:      curBatchNum,
+		BlockHeight:      ctx.BlockHeight(),
+		DataResultRoot:   dataRootHex,
+		ValidatorRoot:    valRootHex,
```

Comment:
Maybe too soon but we ultimately want a single root hash right? With the previous batch root as 1 direct child

Reply:
I don't think we reached an agreement on what exactly to have the validators sign. My understanding was that we want to have them sign the serialized batches, which would also include batch number, block height, etc. Having a single root would facilitate proving, though.

Reply:
We should aim to single root to be signed and not multiple. We should quickly agree on a tree structure. Let's discuss this. 💬

Reply:
Didn't we agree to have a single root?